File: Java/TwoSum.java
/*
Two Sum

Given an array of integers nums and an integer target, return indices of the two numbers such that they add up to target.
*/

class Solution_two_sum {
    public int[] twoSum(int[] nums, int target) {
        int[] found = new int[2];

        for (int i = 0; i < nums.length; i++){
            for (int j = i + 1; j < nums.length; j++){
                if ((nums[i] + nums[j]) == target){
                    found[0] = i;
                    found[1] = j;
                }
            }
        }
        return found;
    }
}

File: LeetCode-Codes/C/TwoSum.c
/*
Two Sum

Given an array of integers nums and an integer target, return indices of the two numbers such that they add up to target.
You may assume that each input would have exactly one solution, and you may not use the same element twice.
*/

int* twoSum(int* nums, int numsSize, int target, int* returnSize) {
    *returnSize = 2;
    int* arr = malloc(2 * (sizeof(int)));
    int count = 0;
    
    for (int i = 0; i < numsSize; i++) {
        for (int j = i + 1; j < numsSize; j++) {
            if ((nums[i] + nums[j]) == target) {
                arr[0] = i;
                arr[1] = j;
            }
        }
    }

    return arr;
}
